README.md

```diff
@@ -1,15 +1,106 @@
+- network.beforeRequestSent
+
+- network.responseCompleted
+- browsingContext.domContentLoaded
+- browsingContext.load
+
+The recorder will not handle subscribing or unsubscribing to those events. For selenium users, the `SeleniumBiDiHarRecorder` can be used to facilitate the process.
+
+```javascript
 const { HarRecorder } = require("bidi-har-export");
 
 const harRecorder = new HarRecorder({ browser: "firefox", version: "111.0a1"});
+
+// It is mandatory to call startRecording before forwarding any event to the
+// recorder.
 harRecorder.startRecording();
-// Forward network and browser context events to the recorder
-harRecorder.recordEvent(evt);
+
+// Forward network and browser context events to the recorder.
+// The recorder expects events as objects containing a "method"
+// and a "params" property.
+harRecorder.recordEvent(beforeRequestSentEvent);
+harRecorder.recordEvent(domContentLoadedEvent);
+harRecorder.recordEvent(loadEvent);
+harRecorder.recordEvent(responseCompletedEvent);
+
+// To complete the recording and generate the HAR record,
+// call stopRecording().
 const harRecord = harRecorder.stopRecording();
+
+// `harRecord` is an object following the HAR specification at
+// https://w3c.github.io/web-performance/specs/HAR/Overview.html
+// It is an object with a single property at the root called `log`
+// See the specification for more details.
+```
+
+### SeleniumBiDiHarRecorder
+
+The `SeleniumBiDiHarRecorder` can be used with a selenium node driver in order to facilitate a recording. This recorder will take care of subscribing and unsubscribing to the necessary event, so that the consumer only has to start and stop the recording.
+
+This recorder is available under the `adapters` namespace.
+
+```javascript
+const fs = require('fs');
+const { BrowsingContext, Builder }  = require('selenium-webdriver');
+const firefox = require('selenium-webdriver/firefox');
+const { adapters } = require('bidi-har-export');
+
+const driver = await new Builder()
+  .forBrowser('firefox')
+  .setFirefoxOptions(new firefox.Options().enableBidi())
+  .build();
+
+const id = await driver.getWindowHandle();
+const browsingContext = await BrowsingContext(driver, { browsingContextId: id });
+
+// The SeleniumBiDiHarRecorder expects a selenium driver object as well as an
+// array of browsing context ids which will be used to know which browsing
+// contexts should be monitored.
+const harRecorder = new adapters.SeleniumBiDiHarRecorder({
+  driver,
+  browsingContextIds: [id],
+});
+
+// Start the recording, perform some navigations (just as an example) and stop
+// the recording.
+await harRecorder.startRecording();
+await browsingContext.navigate('https://example.com', 'complete');
+await browsingContext.navigate('https://wikipedia.org', 'complete');
+const harExport = await harRecorder.stopRecording();
+
+// Save the HAR data to a .har file
+const harData = JSON.stringify(harExport, null, "  ");
```

Comment:
Given that would have to be done by any consumer anyway I wonder if we could move that into `stopRecording()` and let the return value be controlled by a flag eg. `asJSON`? 

Otherwise we could have a `export()` method on the `HARRecorder` class that expects a filename?

Reply:
I think adding options to stopRecording makes sense. I feel like some consumers might still want to get the object and process it before saving it as a file

Reply:
Right. Otherwise we could provide an `export()` method as well. But I don't mind seeing this now for an experimental project.

Reply:
Filed https://github.com/firefox-devtools/bidi-har-export/issues/5